systems/node/api-gateway/pkg/rest/router_test.go

```diff
@@ -0,0 +1,136 @@
+package rest
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+	"github.com/ukama/ukama/systems/common/providers"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	cconfig "github.com/ukama/ukama/systems/common/config"
+	crest "github.com/ukama/ukama/systems/common/rest"
+	"github.com/ukama/ukama/systems/node/api-gateway/pkg"
+	"github.com/ukama/ukama/systems/node/api-gateway/pkg/client"
+	cpb "github.com/ukama/ukama/systems/node/controller/pb/gen"
+	nmocks "github.com/ukama/ukama/systems/node/controller/pb/gen/mocks"
+)
+
+var defaultCors = cors.Config{
+	AllowAllOrigins: true,
+}
+
+var routerConfig = &RouterConfig{
+	serverConf: &crest.HttpConfig{
+		Cors: defaultCors,
+	},
+	auth: &cconfig.Auth{
+		AuthAppUrl:    "http://localhost:4455",
+		AuthServerUrl: "http://localhost:4434",
+		AuthAPIGW:     "http://localhost:8080",
+	},
+}
+
+var testClientSet *Clients
+
+func init() {
+	gin.SetMode(gin.TestMode)
+	testClientSet = NewClientsSet(&pkg.GrpcEndpoints{
+		Timeout:    1 * time.Second,
+		Controller: "0.0.0.0:9092",
+	})
+}
+func TestPingRoute(t *testing.T) {
+	// arrange
+	w := httptest.NewRecorder()
+	arc := &providers.AuthRestClient{}
+	req, _ := http.NewRequest("GET", "/ping", nil)
+	r := NewRouter(testClientSet, routerConfig, arc.MockAuthenticateUser).f.Engine()
+	// act
+	r.ServeHTTP(w, req)
+
+	// assert
+	assert.Equal(t, 200, w.Code)
+	assert.Contains(t, w.Body.String(), "pong")
+}
+
+func Test_RestarteNode(t *testing.T) {
+	// arrange
+	w := httptest.NewRecorder()
+	req, _ := http.NewRequest("POST", "/v1/controllers", nil)
+	arc := &providers.AuthRestClient{}
+	c := &nmocks.ControllerServiceClient{}
+
+	c.On("RestartNode", mock.Anything, mock.Anything).Return(&cpb.RestartNodeResponse{
+		Status: cpb.RestartStatus_RESTART_STATUS_SUCCESS},
+		nil)
+
+	r := NewRouter(&Clients{
+		Controller: client.NewControllerFromClient(c),
+	}, routerConfig, arc.MockAuthenticateUser).f.Engine()
+
+	// act
+	r.ServeHTTP(w, req)
+
+	// assert
+	assert.Equal(t, http.StatusOK, w.Code)
+	c.AssertExpectations(t)
+}
+
+// func TestRouter_PingRoute(t *testing.T) {
```

Comment:
Remove unwanted  code. Missing test case for restart site

Reply:
More test added and legacy code removed.